entity-service/src/main/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationRepository.kt

```diff
@@ -416,4 +416,106 @@ class Neo4jAuthorizationRepository(
     private fun List<Map<String, URI>>?.valuesForRight(accessRight: AccessRight): List<URI>? =
         this?.filter { it.containsKey(accessRight.attributeName) }
             ?.map { it.getValue(accessRight.attributeName) }
+
+    fun getGroups(groupsIds: Set<URI>, offset: Int, limit: Int): Pair<Int, List<Group>> {
+        val matchAuthorizedGroupsClause =
+            """
+            MATCH (group:`$GROUP_TYPE`)
+            WHERE group.id IN ${'$'}groupsIds
+            MATCH (group)-[:HAS_VALUE]->(p:Property { name:"$NGSILD_NAME_PROPERTY" })
+            """.trimIndent()
+
+        val pagingClause = if (limit == 0)
+            """
+            RETURN count(distinct(group)) as count
+            """.trimIndent()
+        else
+            """
+            WITH group, p
+            ORDER BY group.id
+            return 
+                collect(distinct({
+                    groupId:group.id, groupName:p.value
+                })) as groups, 
+                count(distinct(group)) as count
+            SKIP $offset LIMIT $limit
+            """.trimIndent()
+
+        val parameters = mapOf(
+            "groupsIds" to groupsIds.toList().toListOfString()
+        )
+
+        val result = neo4jClient
+            .query(
+                """
+                $matchAuthorizedGroupsClause
+                $pagingClause
+                """
+            )
+            .bindAll(parameters)
+            .fetch()
+            .all()
+
+        return if (limit == 0)
+            Pair((result.firstOrNull()?.get("count") as Long?)?.toInt() ?: 0, emptyList())
```

Comment:
returned count can be null? 🤔

Reply:
If there is no group ?

Reply:
`count(distinct(group))` will return 0?

Reply:
Yes, it return 0. I do it also for the endpoint entities?

Reply:
ok